src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -348,18 +348,42 @@ internal static ImmutableArray<SyntaxReference> GetDeclaringSyntaxReferenceHelpe
             foreach (Location location in locations)
             {
                 // Location may be null. See https://github.com/dotnet/roslyn/issues/28862.
-                if (location == null)
+                if (location == null || !location.IsInSource)
                 {
                     continue;
                 }
-                if (location.IsInSource)
+
+                if (location.SourceSpan.Length != 0)
                 {
-                    SyntaxToken token = (SyntaxToken)location.SourceTree.GetRoot().FindToken(location.SourceSpan.Start);
+                    SyntaxToken token = location.SourceTree.GetRoot().FindToken(location.SourceSpan.Start);
                     if (token.Kind() != SyntaxKind.None)
                     {
                         CSharpSyntaxNode node = token.Parent.FirstAncestorOrSelf<TNode>();
                         if (node != null)
+                        {
                             builder.Add(node.GetReference());
+                        }
+                    }
+                }
+                else
+                {
+                    // Since the location we're interested in can't contain a token, we'll inspect the whole tree,
+                    // pruning away branches that don't contain that location. We'll pick the narrowest node of the type
+                    // we're looking for.
+                    // eg: finding the ParameterSyntax from the empty location of a blank identifier
+                    SyntaxNode parent = location.SourceTree.GetRoot();
+                    SyntaxNode found = null;
+                    foreach (var descendant in parent.DescendantNodesAndSelf(c => c.Location.SourceSpan.Contains(location.SourceSpan)))
+                    {
+                        if (descendant is TNode && descendant.Location.SourceSpan.Contains(location.SourceSpan))
```

Comment:
>descendant.Location.SourceSpan.Contains(location.SourceSpan) [](start = 51, length = 60)

Why do we have to do this check again?

Reply:
I think this can be simplified. I'll do that in next records PR.

Reply:
It's indeed possible to simplify (added that to [upcoming PR for ToString](https://github.com/dotnet/roslyn/pull/46616)). Thanks for catching that

Reply:
Never mind. This cannot be removed (crashes multiple tests).
The lambda in `DescendantNodesAndSelf` is not to decide which children to explore, but which parents to get children from.